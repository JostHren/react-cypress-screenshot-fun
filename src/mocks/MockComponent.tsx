/* eslint-disable @typescript-eslint/no-unsafe-member-access */
import { inspect } from 'util';

import { ReactElement } from 'react';

export const MockComponent = ({ children, componentName, testId, ...props }: any): ReactElement => (
  <div data-component-name={componentName} data-testid={testId} onClick={props?.onClick}>
    --- PROPS
    {inspect(props)}
    --- CHILDREN
    {children}
  </div>
);

export const mockComponent = (
  componentName: string,
  testId?: string,
): Record<string, (props: any) => ReactElement> => ({
  [componentName]: (props: any) => (
    <MockComponent componentName={componentName} testId={testId} {...props} />
  ),
});
