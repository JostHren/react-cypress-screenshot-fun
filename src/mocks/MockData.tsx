export const MOCKED_FLOWER = {
  id: 'mock-id',
  name: 'mock-name',
  latinName: 'mock-latin-name',
  genus: 'mock-genus',
  pictureUrl: 'mock-picture-url',
  authorId: 'mock-author-id',
  sightingsNum: 999,
  isFavorite: false,
  description: 'mock-description',
  features: 'mock-features',
};

export const MOCKED_FLOWERS = {
  items: [MOCKED_FLOWER],
  order: [],
  filters: {},
  total: 99,
  page: 1,
  limit: 9,
};
