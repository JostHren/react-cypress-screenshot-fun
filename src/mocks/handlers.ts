// src/mocks/handlers.js
import { http, HttpResponse } from 'msw'
import { MOCKED_DATA, MOCKED_TOKEN, MOCKED_USER } from './MockData'


const API_BASE = import.meta.env.VITE_API_URL
 
export const handlers = [
  http.post(`${API_BASE}account/register`, () => {
    return HttpResponse.json(MOCKED_TOKEN)
  }),
  http.post(`${API_BASE}account/login`, () => {
    return HttpResponse.json(MOCKED_TOKEN)

  }),
  http.post(`${API_BASE}account/refresh`, () => {
    return HttpResponse.json(MOCKED_TOKEN)

  }),
  http.get(`${API_BASE}account/me`, () => {
    return HttpResponse.json(MOCKED_USER)

  }),
  http.get(`${API_BASE}account/birds`, () => {
    return HttpResponse.json(MOCKED_DATA)
  }),
  http.get(`${API_BASE}birds`, () => {
    return HttpResponse.json(MOCKED_DATA)
  }),
]