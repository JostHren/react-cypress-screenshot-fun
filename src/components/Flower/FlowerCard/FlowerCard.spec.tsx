import { render } from '@testing-library/react';

import { FlowerCard } from './FlowerCard';
import { MOCKED_FLOWER } from '../../../mocks/MockData';

describe('Flower component', () => {
  it('Flower component to render', () => {
    const { container } = render(<FlowerCard flower={MOCKED_FLOWER} />);

    expect(container).toMatchSnapshot();
  });

  it('Flower component should render if flower is favorite', () => {
    const { container } = render(<FlowerCard flower={{ ...MOCKED_FLOWER, isFavorite: true }} />);

    expect(container).toMatchSnapshot();
  });
});
