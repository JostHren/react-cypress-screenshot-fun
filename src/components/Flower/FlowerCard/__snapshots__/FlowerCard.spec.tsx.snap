// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Flower component > Flower component should render if flower is favorite 1`] = `
<div>
  <div
    data-component-name="Box"
  >
    --- PROPS
    {
  sx: {
    position: 'relative',
    borderRadius: 1,
    width: '100%',
    aspectRatio: '28/35',
    backgroundColor: 'lightgrey',
    backgroundImage: 'url(mock-picture-url)',
    backgroundRepeat: 'no-repeat',
    backgroundSize: 'cover'
  }
}
    --- CHILDREN
    <div
      data-component-name="Box"
    >
      --- PROPS
      {
  position: 'absolute',
  right: '20px',
  top: '20px',
  bgcolor: '#EAA79E',
  width: '30px',
  height: '30px',
  borderRadius: 10,
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'space-around'
}
      --- CHILDREN
      <div
        data-component-name="SvgIcon"
      >
        --- PROPS
        { sx: { height: '12px' } }
        --- CHILDREN
        <svg
          fill="none"
          height="12"
          viewBox="0 0 13 12"
          width="13"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M12.7184 5.43853C12.9739 5.19714 13.064 4.84571 12.9537 4.51748C12.8431 4.18925 12.5557 3.95538 12.2011 3.90647L9.05377 3.4657C8.91905 3.44595 8.80354 3.36538 8.74366 3.24876L7.33663 0.500967C7.17881 0.192173 6.85731 0 6.50034 0C6.14305 0 5.82286 0.192173 5.66374 0.500967L4.2567 3.24876C4.19683 3.36538 4.08131 3.4472 3.94659 3.4657L0.799306 3.90647C0.444619 3.95538 0.155987 4.18925 0.0456762 4.51748C-0.0636588 4.84571 0.0264775 5.19839 0.281918 5.43853L2.55973 7.57657C2.65637 7.66843 2.70095 7.79947 2.67785 7.928L2.14029 10.9479C2.09343 11.2138 2.16567 11.4733 2.34366 11.6765C2.62221 11.995 3.1064 12.0919 3.49396 11.8947L6.30836 10.4692C6.42648 10.409 6.57388 10.4103 6.69201 10.4692L9.50771 11.8947C9.64372 11.9643 9.78983 12 9.94114 12C10.2158 12 10.4774 11.8824 10.6567 11.6765C10.8347 11.4733 10.9073 11.2138 10.8601 10.9479L10.3225 7.928C10.2994 7.79947 10.344 7.66843 10.4406 7.57657L12.7184 5.43853Z"
            fill="#D4D8D9"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </div>
    <div
      data-component-name="Stack"
    >
      --- PROPS
      {
  direction: 'column',
  sx: {
    height: '100%',
    justifyContent: 'flex-end',
    alignItems: 'center',
    paddingBottom: '20px'
  }
}
      --- CHILDREN
      <div
        data-component-name="Typography"
      >
        --- PROPS
        { noWrap: true, variant: 'h4', color: 'white' }
        --- CHILDREN
        mock-name
      </div>
      <div
        data-component-name="Typography"
      >
        --- PROPS
        { noWrap: true, variant: 'h5', color: 'white', mb: '20px' }
        --- CHILDREN
        mock-latin-name
      </div>
      <div
        data-component-name="Chip"
      >
        --- PROPS
        {
  sx: { background: 'red' },
  label: '999 sightings',
  variant: 'filled',
  color: 'primary'
}
        --- CHILDREN
      </div>
    </div>
  </div>
</div>
`;

exports[`Flower component > Flower component to render 1`] = `
<div>
  <div
    data-component-name="Box"
  >
    --- PROPS
    {
  sx: {
    position: 'relative',
    borderRadius: 1,
    width: '100%',
    aspectRatio: '28/35',
    backgroundColor: 'lightgrey',
    backgroundImage: 'url(mock-picture-url)',
    backgroundRepeat: 'no-repeat',
    backgroundSize: 'cover'
  }
}
    --- CHILDREN
    <div
      data-component-name="Box"
    >
      --- PROPS
      {
  position: 'absolute',
  right: '20px',
  top: '20px',
  bgcolor: 'white',
  width: '30px',
  height: '30px',
  borderRadius: 10,
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'space-around'
}
      --- CHILDREN
      <div
        data-component-name="SvgIcon"
      >
        --- PROPS
        { sx: { height: '12px' } }
        --- CHILDREN
        <svg
          fill="none"
          height="12"
          viewBox="0 0 13 12"
          width="13"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M12.7184 5.43853C12.9739 5.19714 13.064 4.84571 12.9537 4.51748C12.8431 4.18925 12.5557 3.95538 12.2011 3.90647L9.05377 3.4657C8.91905 3.44595 8.80354 3.36538 8.74366 3.24876L7.33663 0.500967C7.17881 0.192173 6.85731 0 6.50034 0C6.14305 0 5.82286 0.192173 5.66374 0.500967L4.2567 3.24876C4.19683 3.36538 4.08131 3.4472 3.94659 3.4657L0.799306 3.90647C0.444619 3.95538 0.155987 4.18925 0.0456762 4.51748C-0.0636588 4.84571 0.0264775 5.19839 0.281918 5.43853L2.55973 7.57657C2.65637 7.66843 2.70095 7.79947 2.67785 7.928L2.14029 10.9479C2.09343 11.2138 2.16567 11.4733 2.34366 11.6765C2.62221 11.995 3.1064 12.0919 3.49396 11.8947L6.30836 10.4692C6.42648 10.409 6.57388 10.4103 6.69201 10.4692L9.50771 11.8947C9.64372 11.9643 9.78983 12 9.94114 12C10.2158 12 10.4774 11.8824 10.6567 11.6765C10.8347 11.4733 10.9073 11.2138 10.8601 10.9479L10.3225 7.928C10.2994 7.79947 10.344 7.66843 10.4406 7.57657L12.7184 5.43853Z"
            fill="#D4D8D9"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </div>
    <div
      data-component-name="Stack"
    >
      --- PROPS
      {
  direction: 'column',
  sx: {
    height: '100%',
    justifyContent: 'flex-end',
    alignItems: 'center',
    paddingBottom: '20px'
  }
}
      --- CHILDREN
      <div
        data-component-name="Typography"
      >
        --- PROPS
        { noWrap: true, variant: 'h4', color: 'white' }
        --- CHILDREN
        mock-name
      </div>
      <div
        data-component-name="Typography"
      >
        --- PROPS
        { noWrap: true, variant: 'h5', color: 'white', mb: '20px' }
        --- CHILDREN
        mock-latin-name
      </div>
      <div
        data-component-name="Chip"
      >
        --- PROPS
        {
  sx: { background: 'rgba(0, 0, 0, 0.5)' },
  label: '999 sightings',
  variant: 'filled',
  color: 'secondary'
}
        --- CHILDREN
      </div>
    </div>
  </div>
</div>
`;
